astropy/utils/data.py

```diff
@@ -914,20 +914,24 @@ def _find_hash_fn(hexdigest, pkgname='astropy'):
     return None
 
 
-def get_free_space_in_dir(path):
+def get_free_space_in_dir(path, return_quantity=False):
     """
     Given a path to a directory, returns the amount of free space (in
     bytes) on that filesystem.
 
     Parameters
     ----------
     path : str
-        The path to a directory
+        The path to a directory.
+
+    return_quantity : bool
```

Comment:
I wondered a bit about this: How about instead asking for the unit someone wants it in? With `None` meaning no unit ( and perhaps in future `True` for the "best" unit).

Reply:
Very nice idea, but I am not sure how to implement that cleanly. There is this:

https://github.com/astropy/astropy/blob/b5d381f8d576a47ad4c873c015b0c913caec878c/astropy/utils/console.py#L440

But it seems to take great length to avoid using `units`. Do you remember why?

Reply:
I'd leave the "True" for another time - I'd probably rewrite human-readable (there's a closed PR somewhere...)

Reply:
Are you thinking about #4373 ?

Reply:
No, I was thinking of #7033 - which generalizes it for all units (well, non-composed ones - which to me is the tricky bit: what does one do with, e.g., velocity).